File: domain/src/main/kotlin/boaz/lol/co/domains/account/Account.java
package boaz.lol.co.domains.account;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class Account {
    private final Long id;
    private final String email;
    private final String password;
    private final String riotName;
    private final String riotTag;
    private final String riotId;
    private final Boolean isAgreeToPersonalInfo;
    private final Boolean isAgreeToEmail;
    private final Boolean isCertificatedEmail;

    @Builder
    public Account(Long id, String email, String password, String riotName, String riotTag, String riotId, Boolean isAgreeToPersonalInfo, Boolean isAgreeToEmail, Boolean isCertificatedEmail) {
        this.id = id;
        this.email = email;
        this.password = password;
        this.riotName = riotName;
        this.riotTag = riotTag;
        this.riotId = riotId;
        this.isAgreeToPersonalInfo = isAgreeToPersonalInfo;
        this.isAgreeToEmail = isAgreeToEmail;
        this.isCertificatedEmail = isCertificatedEmail;
    }
}

File: domain/src/main/kotlin/boaz/lol/co/domains/account/Account.kt
package boaz.lol.co.domains.account

data class Account(
    val id: Long,
    var email: String,
    val password: String,
    var riotName: String,
    var riotTag: String,
    var riotId: String,
    val isAgreeToPersonalInfo: Boolean,
    val isAgreeToEmail: Boolean,
    val isCertificatedEmail: Boolean
) {
    fun to() : AccountData {
        return AccountData(id, email, riotId)
    }

    fun update(accountModify: AccountModify) {
        this.email = accountModify.email
        this.riotName = accountModify.riotName
        this.riotTag = accountModify.riotTag
        this.riotId = accountModify.riotId
    }
}